backend/api/quivr_api/modules/sync/utils/dropboxutils.py

```diff
@@ -0,0 +1,338 @@
+
+                metadata, file_data = dbx.files_download(file_id)  # type: ignore
+                # logger.debug("🔥 Filedata :", file_data.content)
+                file_data = BytesIO(file_data.content)
+
+                # Check if the file already exists in the storage
+                if check_file_exists(brain_id, file_name):
+                    logger.debug("🔥 File already exists in the storage: %s", file_name)
+
+                    self.storage.remove_file(brain_id + "/" + file_name)
+                    BrainsVectors().delete_file_from_brain(brain_id, file_name)
+
+                # Check if the file extension is compatible
+                if file_name.split(".")[-1] not in [
+                    "pdf",
+                    "txt",
+                    "md",
+                    "csv",
+                    "docx",
+                    "xlsx",
+                    "pptx",
+                    "doc",
+                ]:
+                    logger.info("File is not compatible: %s", file_name)
+                    continue
+
+                to_upload_file = UploadFile(
+                    file=file_data,
+                    filename=file_name,
+                )
+
+                # Check if the file already exists in the database
+                existing_files = self.sync_files_repo.get_sync_files(sync_active_id)
+                existing_file = next(
+                    (f for f in existing_files if f.path == file_name), None
+                )
+
+                supported = False
+                if (existing_file and existing_file.supported) or not existing_file:
+                    supported = True
+                    await upload_file(
+                        to_upload_file,
+                        brain_id,
+                        current_user,
+                        bulk_id,
+                        "DropBox",
+                        file.web_view_link,
+                        notification_id=file.notification_id,
+                    )
+
+                if existing_file:
+                    # Update the existing file record
+                    self.sync_files_repo.update_sync_file(
+                        existing_file.id,
+                        SyncFileUpdateInput(
+                            last_modified=modified_time,
+                            supported=supported,
+                        ),
+                    )
+                else:
+                    # Create a new file record
+                    self.sync_files_repo.create_sync_file(
+                        SyncFileInput(
+                            path=file_name,
+                            syncs_active_id=sync_active_id,
+                            last_modified=modified_time,
+                            brain_id=brain_id,
+                            supported=supported,
+                        )
+                    )
+
+                downloaded_files.append(file_name)
+
+            except Exception as error:
```

Comment:
Could probably divide function into two.The Exception we catch is too generic in my sense and the block after the exception has a lot of logic that could be not related to the exception raised.  If the logic after the exception can be move before doing any logic, maybe it's better to have it upfront.